sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -17,9 +17,7 @@ public abstract partial class AsyncPageable<T> : System.Collections.Generic.IAsy
     }
     public static partial class AzureCoreExtensions
     {
-        public static dynamic ToDynamicFromJson(this System.BinaryData utf8Json) { throw null; }
-        public static dynamic ToDynamicFromJson(this System.BinaryData utf8Json, Azure.Core.Dynamic.DynamicCaseMapping caseMapping, Azure.Core.Dynamic.DynamicDateTimeHandling dateTimeHandling = Azure.Core.Dynamic.DynamicDateTimeHandling.Rfc3339) { throw null; }
-        public static dynamic ToDynamicFromJson(this System.BinaryData utf8Json, Azure.Core.Dynamic.DynamicDataOptions options) { throw null; }
+        public static dynamic ToDynamicFromJson(this System.BinaryData utf8Json, Azure.Core.Dynamic.PropertyNameLookup existingPropertyLookup, Azure.Core.Serialization.PropertyNameConversion newPropertyConversion = Azure.Core.Serialization.PropertyNameConversion.None, Azure.Core.Dynamic.DynamicDateTimeHandling dateTimeHandling = Azure.Core.Dynamic.DynamicDateTimeHandling.Rfc3339) { throw null; }
```

Comment:
I don't think we need to call the first parameter "existingPropertyLookup". "propertyLookup" would be clear enough.

Reply:
I thought we won't be using an enums for the DateTime  format

Reply:
BTW, another implication of this design is that users who write and want a mapping, will forever have to specify arguments. i.e. even if we add "CaseInsensitive" default to the lookup enum, the user who wants recasing on write will have to do:
```
dynamic json = data.ToDynamicFromJson(PropertyNameLookup.CaseInsensitive, PropertyNameConversion.CamelCase);
```

Reply:
> I don't think we need to call the first parameter "existingPropertyLookup". "propertyLookup" would be clear enough.

I'm happy to make that change.

Reply:
> I thought we won't be using an enums for the DateTime format

I'm tracking the work to rewrite that API as a separate feature.  I do plan to address it in a separate PR.

Reply:
> BTW, another implication of this design is that users who write and want a mapping, will forever have to specify arguments. i.e. even if we add "CaseInsensitive" default to the lookup enum, the user who wants recasing on write will have to do:

Why not?  Wouldn't it be possible to add an overload that takes PropertyNameConversion as the first parameter, or are there design guidelines against doing that?

Reply:
Yes, there are design guidelines stating that overloads should not reorder parameters.

Reply:
Oh, that is worse than I thought.

Reply:
I think two enums, but one enum for this method and one different for the other solves all problems that people had.